### DISCUSSION TITLE
GitHub rate limit while using docker

### DISCUSSION THREAD

**Discussion by JaneX8** [projectdiscovery/nuclei#2985] [Q&A] [answered]
Once I was testing around and didn't realize an update was requested on every test run, so I hit the rate limit. I think that in general updating automatically and explicitly disabling that with `-duc` is the right thing to do however that's annoying when there is rate limiting applied.

I suggest to either increase the rate limit or even better use a local update caching (simple last datetime update check) for example limiting the rate to once per hour or day. Additionally a note could be given to the user when the update was actually performed or when it was skipped. Another flag could potentially disable or overrule this "local caching" behavior. So, basically it's client-side rate limiting, while keeping the auto updates enabled by default.

**Comment by ehsandeep** [comment]
@JaneX8 Could you clarify from where you got rate-limited? as the check is done from version check API at https://version-check.nuclei.sh/versions and GitHub get called only when you are running outdated version, so it should only once.

**Comment by JaneX8** [comment]
Unfortunately this happened before and I didn't log event details. It could be indeed GitHub rate limiting. I was running nuclei from within a docker container. Version: 2.7.9.

**Comment by Mzack9999** [comment]
The template check seems to increases the startup time consistently. Limiting it to once every few hours seems reasonable. What do you think @ehsandeep ?

**Comment by ehsandeep** [comment, marked as answer]
@JaneX8 Probably rate limit got triggered because of using fresh docker for all the runs. basically, it means it installs nuclei templates for each run and to avoid rate limit in this situation, we have added `GITHUB_TOKEN` env variable support that can be used along with docker and not hit rate limit as GitHub have strict limitations on public API.

**Comment by JaneX8** [comment]
@ehsandeep you are right. I did use `docker run --rm` while testing.
